### PR TITLE
moved method to store item position from CGUIMediaWindow to CGUIWindo…

### PR DESCRIPTION
**Pull request by wiromare**
…wVideoBase

part 2 of the previous PR #8873

### CODE REVIEW

File: xbmc/video/windows/GUIWindowVideoBase.cpp [hunk 1]
```diff
@@ -118,6 +118,8 @@ bool CGUIWindowVideoBase::OnMessage(CGUIMessage& message)
     if (m_thumbLoader.IsLoading())
       m_thumbLoader.StopThread();
     m_database.Close();
+    // store the selected item
+    SetSelectedItem(m_vecItems->GetPath(), m_viewControl.GetSelectedItem());
```
Comment (xhaggi) [inline xbmc/video/windows/GUIWindowVideoBase.cpp]: please put it in OnDeinitWindow() like I said before.

Comment (wiromare) [inline xbmc/video/windows/GUIWindowVideoBase.cpp]: ok

**Comment by Razzeee** [comment]
What about music?

**Comment by xhaggi** [comment]
@Razzeee we need to add the same to `CGUIWindowMusicBase`

**Comment by Paxxi** [comment]
Do I understand this correctly that we're moving the code from guumediawindow to videobase for some reason and would then duplicate the code for music? Logic seems fine but don't move shared code down in the inheritance chain without good reason.

**Comment by xhaggi** [comment]
As we only need this kind of abstraction for video and music we should not do it in our media base class `CGUIMediaWindow`.

File: xbmc/video/windows/GUIWindowVideoBase.h [hunk 2]
```diff
@@ -43,6 +43,8 @@ class CGUIWindowVideoBase : public CGUIMediaWindow, public IBackgroundLoaderObse
   virtual ~CGUIWindowVideoBase(void);
   virtual bool OnMessage(CGUIMessage& message) override;
   virtual bool OnAction(const CAction &action) override;
+  virtual void OnInitWindow() override;
+  virtual void OnDeinitWindow(int nextWindowID) override;
```
Comment (Paxxi) [inline xbmc/video/windows/GUIWindowVideoBase.h]: You don't need to specify virtual when using override, override implies virtual. You don’t need to change the pr, just meant as a tip for the future :)

**Comment by Paxxi** [comment]
What else inherits from mediawindow? On my phone so can't really check, also why wouldn't this make sense for pictures as well?

**Comment by xhaggi** [comment]
eventlog, addonbrowser, pvr etc... for pictures where do you see the advantage?

**Comment by Paxxi** [comment]
Well anything that has a directory navigation benefits from storing the selected path, addon browser as well if going back and forth between categories, not sure about eventlog but it feels like that shouldn't inherit from mediawindow

**Comment by xhaggi** [comment]
I'm not sure if you understand what's done here. we already store the last selected position if you browse within e.g. addon directory back/forward. this is possible for ages. he want to separate this behaviour within one window depending on the selected media type e.g. video, tvshows etc.

**Comment by Paxxi** [comment]
I may have misunderstood but looking at the changes it doesn't care about media type it stores selection per path which is a valid thing for the windows I mentioned.

**Comment by xhaggi** [comment]
Yep that's wrong and the more I think about it the more this is wrong ;)

File: xbmc/video/windows/GUIWindowVideoBase.h (hunk 2 above)
Comment (wiromare) [inline xbmc/video/windows/GUIWindowVideoBase.h]: ok thx,

but there are alot of virtual ... override 
within kodi code :)

**Comment by wiromare** [comment]
I have to add these changes also in CGUIWindowMusicBase?

File: xbmc/video/windows/GUIWindowVideoBase.h (hunk 2 above)
Comment (Montellese) [inline xbmc/video/windows/GUIWindowVideoBase.h]: @Paxxi: Is that 100% accurate? If you have another class inherit from this class and want them to be able to override the same method you still need to put the `virtual` there or not? In this case there's `CGUIWindowVideoNav` which derives from `CGUIWindowVideoBase` which derives from `CGUIMediaWindow` and so on. For `CGUIWindowVideoNav` to be able to properly override `OnInitWindow()` et al. you need to define them virtual in `CGUIWindowVideoBase`.

**Pull request closed without merging**